src/NServiceBus.Core/Audit/AuditAction.cs

```diff
@@ -0,0 +1,17 @@
+﻿namespace NServiceBus.Audit
+{
+    using System;
+    using System.Collections.Generic;
+    using Pipeline;
+
+    /// <summary>
+    /// Base class for audit actions.
+    /// </summary>
+    public abstract class AuditAction
+    {
+        /// <summary>
+        /// Gets the messages, if any, this audit operation should result in.
+        /// </summary>
+        public abstract IEnumerable<IRoutingContext> GetRoutingContexts(IAuditContext context, TimeSpan? timeToBeReceived);
```

Comment:
On the other PR we will return here `IReadOnlyCollection`

Reply:
We also have a dedicated interface that we pass into `Get...Contexts` that exposes the metadata as non mutable